webapp/channels/src/components/dialog_router/interactive_dialog_adapter.tsx

```diff
@@ -0,0 +1,373 @@
+
+            if (result?.error) {
+                this.logError('Failed to notify server of dialog cancellation', {
+                    error: result.error,
+                    callbackId: this.props.callbackId,
+                    url: this.props.url,
+                });
+            }
+        } catch (error) {
+            this.logError('Failed to notify server of dialog cancellation', {
+                error: error instanceof Error ? error.message : String(error),
+                callbackId: this.props.callbackId,
+                url: this.props.url,
+            });
+        }
+
+        // Don't throw here as cancellation should always succeed from UX perspective
+    };
+
+    /**
+     * No-op lookup adapter for unsupported legacy feature
+     */
+    private performLookupCall = async (): Promise<DoAppCallResult<unknown>> => {
+        this.logWarn('Unexpected lookup call in Interactive Dialog adapter - this should not happen');
+        return {
+            data: {
+                type: 'ok' as const,
+                data: {items: []},
+            },
+        };
+    };
+
+    /**
+     * No-op refresh adapter for unsupported legacy feature
+     */
+    private refreshOnSelect = async (): Promise<DoAppCallResult<unknown>> => {
+        this.logWarn('Unexpected refresh call in Interactive Dialog adapter - this should not happen');
+        return {
+            data: {
+                type: 'ok' as const,
+            },
+        };
+    };
+
+    /**
+     * No-op ephemeral response adapter for legacy compatibility
+     */
+    private postEphemeralCallResponseForContext = (): void => {
+    };
+
+    render() {
+        const {form, error} = this.convertToAppForm();
+
+        if (error) {
+            this.logError('Failed to convert dialog to app form', error);
+            return null;
+        }
+
+        if (!form) {
+            this.logError('No form generated from dialog conversion');
+            return null;
+        }
+
+        // Create a minimal context for legacy interactive dialogs
+        const context = createCallContext(
+            'legacy-interactive-dialog', // app_id for legacy dialogs
+            'interactive_dialog', // location
+        );
+
+        return (
+            <DynamicAppsFormContainer
+                form={form}
+                context={context}
+                onExited={this.props.onExited || (() => {})}
+                onHide={this.cancelAdapter}
+                actions={{
+                    doAppSubmit: this.submitAdapter,
+                    doAppFetchForm: this.refreshOnSelect,
+                    doAppLookup: this.performLookupCall,
+                    postEphemeralCallResponseForContext: this.postEphemeralCallResponseForContext,
+                }}
+            />
+        );
+    }
+}
+
+// Dynamic wrapper component for AppsFormContainer to avoid circular dependency
+const DynamicAppsFormContainer: React.FC<any> = (props) => {
```

Comment:
One last thing here. I realized that we actually have our `makeAsyncComponent` utility for lazy loading components that we could reuse here instead of writing custom logic for it. I tried it out locally, and it seemed to work same as this.

https://github.com/mattermost/mattermost/commit/950b73cfd19edee2866a2c2545075007383d2855

Reply:
Nice...that's what I needed...